src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+    const BinaryPredicateExpression& expression) {
+  Assert(expression.predicate_condition == PredicateCondition::Like ||
+             expression.predicate_condition == PredicateCondition::NotLike,
+         "Expected PredicateCondition Like or NotLike");
+
+  const auto left_results = evaluate_expression_to_result<std::string>(*expression.left_operand());
+  const auto right_results = evaluate_expression_to_result<std::string>(*expression.right_operand());
+
+  const auto invert_results = expression.predicate_condition == PredicateCondition::NotLike;
+
+  const auto result_size = _result_size(left_results->size(), right_results->size());
+  auto result_values = std::vector<int32_t>(result_size, 0);
+
+  /**
+   * Three different kinds of LIKE are considered for performance reasons and avoid redundant creation of the
+   * LikeMatcher
+   *    - `a LIKE b`
+   *    - `a LIKE '%hello%'`
+   *    - `'hello' LIKE b`
+   */
+  if (left_results->is_literal() == right_results->is_literal()) {
+    // E.g., `a LIKE b` - A new matcher for each row and a different value as well
+    for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+      LikeMatcher{right_results->values[row_idx]}.resolve(invert_results, [&](const auto& matcher) {
+        result_values[row_idx] = matcher(left_results->values[row_idx]);
+      });
+    }
+  } else if (!left_results->is_literal() && right_results->is_literal()) {
+    // E.g., `a LIKE '%hello%'` -- A single matcher for all rows
+    LikeMatcher like_matcher{right_results->values.front()};
+
+    for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+      like_matcher.resolve(invert_results, [&](const auto& matcher) {
+        result_values[row_idx] = matcher(left_results->values[row_idx]);
+      });
+    }
+  } else {
+    // E.g., `'hello' LIKE b` -- A new matcher for each row but the value to check is constant
+    for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+      LikeMatcher{right_results->values[row_idx]}.resolve(
+          invert_results, [&](const auto& matcher) { result_values[row_idx] = matcher(left_results->values.front()); });
+    }
+  }
+
+  auto result_nulls = _evaluate_default_null_logic(left_results->nulls, right_results->nulls);
+
+  return std::make_shared<ExpressionResult<int32_t>>(std::move(result_values), std::move(result_nulls));
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_like_expression(
+    const BinaryPredicateExpression& expression) {
+  Fail("Can only evaluate predicates to int32_t (aka bool)");
+}
+
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_is_null_expression<int32_t>(
+    const IsNullExpression& expression) {
+  std::vector<int32_t> result_values;
+
+  _resolve_to_expression_result_view(*expression.operand(), [&](const auto& view) {
```

Comment:
I am not sure how to do this, but it would be good to have more documentation for concepts like result views etc.

Reply:
I put these views into their own file (expression_result_views.hpp) and wrote a bit about them.